c/driver/postgresql/postgres_type.h

```diff
@@ -223,6 +223,14 @@ class PostgresType {
         NANOARROW_RETURN_NOT_OK(ArrowSchemaSetType(schema, NANOARROW_TYPE_BINARY));
         break;
 
+      case PostgresTypeId::kTimestamp:
+        // TODO: it's possible to set a precision in the type, but how do we get that
```

Comment:
I think this is always correct. You can set a precision with the timestamp, but it doesn't change the underlying storage type or resolution

https://www.postgresql.org/docs/current/datatype-datetime.html

Reply:
If you cared to see what `p` is for a given column I think it maps to `atttypmod` in pg_attribute, much like the length specifier of VARCHAR:

https://www.postgresql.org/docs/current/catalog-pg-attribute.html

Reply:
Ah, thanks.

Yeah, when I was checking the type mapping for Postgres on the Java side I decided to make it vary based on the specified precision, but just always using microsecond would be easiest, so I'll fix up the Java side after this.

That said we probably do need to figure this out for TIMESTAMPTZ and eventually things like decimals.

Reply:
Makes sense. Looks like libpq has the `PQfmod` function to inspect result column types, so could definitely build an abstraction on top of that alongside `PQftype` to handle this more generically